### PR TITLE
`ultralytics 8.3.94` Limit Segment val plots to 50 items/image for speed

### LINKED ISSUE #19790: In the validation images, there are many objects that are not detected
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Val

### Bug

I trained a model to detect berries. I consider the dataset is very representative: around 4k images of hight-resolution with a lot of variability. In general, the object to be detected appears multiple times in each image, averaging around 30-50 instances per image. That is, many instances of different sizes and locations. After training this model, the metrics indicate that the model performs very well. However, when reviewing the test images, I notice that several objects are not detected (it detects about 80% of the objects).

My question is: how can I solve this? Is the model appropriate (YOLOv11)? Should I try another model or adjust some parameters?

btw, I have trained this model (`yolo11s` and `yolo11n`) several times but I still getting the same results

### Environment

```
Ultralytics 8.3.90 🚀 Python-3.12.3 torch-2.5.1 CUDA:0 (NVIDIA GeForce RTX 4070 Ti SUPER, 16376MiB)
Setup complete ✅ (24 CPUs, 95.2 GB RAM, 1804.4/2725.3 GB disk)

OS                  Windows-11-10.0.26100-SP0
Environment         Windows
Python              3.12.3
Install             pip
Path                C:\Users\Josue\miniconda3\envs\gpu\lib\site-packages\ultralytics
RAM                 95.19 GB
Disk                1804.4/2725.3 GB
CPU                 AMD Ryzen 9 7900X3D 12-Core Processor
CPU count           24
GPU count           1
CUDA                12.4

numpy               ✅ 1.26.4<=2.1.1,>=1.23.0
matplotlib          ✅ 3.9.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.0>=1.4.1
torch               ✅ 2.5.1>=1.8.0
torch               ✅ 2.5.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0
{'OS': 'Windows-11-10.0.26100-SP0', 'Environment': 'Windows', 'Python': '3.12.3', 'Install': 'pip', 'Path': 'C:\\Users\\Josue\\miniconda3\\envs\\gpu\\lib\\site-packages\\ultralytics', 'RAM': '95.19 GB', 'Disk': '1804.4/2725.3 GB', 'CPU': 'AMD Ryzen 9 7900X3D 12-Core Processor', 'CPU count': 24, 'GPU': 'NVIDIA GeForce RTX 4070 Ti SUPER, 16376MiB', 'GPU count': 1, 'CUDA': '12.4', 'Package Info': {'numpy': '✅ 1.26.4<=2.1.1,>=1.23.0', 'matplotlib': '✅ 3.9.0>=3.3.0', 'opencv-python': '✅ 4.10.0.84>=4.6.0', 'pillow': '✅ 10.4.0>=7.1.2', 'pyyaml': '✅ 6.0.1>=5.3.1', 'requests': '✅ 2.32.3>=2.23.0', 'scipy': '✅ 1.14.0>=1.4.1', 'torch': '✅ 2.5.1!=2.4.0,>=1.8.0; sys_platform == "win32"', 'torchvision': '✅ 0.20.1>=0.9.0', 'tqdm': '✅ 4.66.5>=4.64.0', 'psutil': '✅ 5.9.0', 'py-cpuinfo': '✅ 9.0.0', 'pandas': '✅ 2.2.2>=1.1.4', 'seaborn': '✅ 0.13.2>=0.11.0', 'ultralytics-thop': '✅ 2.0.0>=2.0.0'}}
```

### Minimal Reproducible Example

Dataset yaml:
```yaml
path: C:/Users/Josue/strawberry/data/datasets/berry_polygon
train: train
val: test

nc: 1
names:
  0: berry

```

Training code:
```python
from ultralytics.data.augment import Albumentations
from ultralytics.engine.trainer import BaseTrainer
import albumentations as A


def callback_custom_albumentations(trainer: BaseTrainer):
    T = [
        A.OneOf([A.ToGray(), A.RandomGamma(), A.CLAHE()]),
        A.OneOf([A.Blur(blur_limit=(3, 9)), A.MedianBlur(blur_limit=(3, 7))]),
        A.RandomBrightnessContrast(brightness_limit=0.2, contrast_limit=0.2, p=0.25),
        A.HueSaturationValue(
            hue_shift_limit=10, sat_shift_limit=10, val_shift_limit=10, p=0.25
        ),
    ]
    transform = A.Compose(T)

    for t in trainer.train_loader.dataset.transforms.transforms:
        if isinstance(t, Albumentations):
            t.transform = transform
            break

    trainer.train_loader.reset()


dataset_path = "../../data/datasets/berry_dataset.yaml"
wandb.init(project=PROJECT, job_type=RUN_NAME, name=RUN_NAME, notes=DESCRIPTION)
model = YOLO("yolo11s-seg.pt")
model.add_callback("on_pretrain_routine_end", callback_custom_albumentations)

results = model.train(
    data=dataset_path,
    project=PROJECT,
    name=RUN_NAME,
    single_cls=True,
    plots=True,
    batch=32,
    epochs=256,
    imgsz=imgsz,
    seed=7,
    degrees=2,
    dropout=0.2,
    mosaic=0.5,
)

wandb.finish()

```

### Additional

Image

Image

Image

Image

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Y-T-G**
Closes #19790

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary  
Improved validation plotting in YOLO segmentation models for better performance and usability. 🚀  

### 📊 Key Changes  
- Increased the limit of plotted masks during validation from 15 to 50 for better visualization.  
- Added a warning message when more than 50 masks are plotted, ensuring users are aware of the limit. ⚠️  
- Updated the maximum detection limit for plotting predictions to 50 for consistency.  

### 🎯 Purpose & Impact  
- **Enhanced Visualization**: Users can now view up to 50 masks during validation, providing more comprehensive insights into model performance.  
- **Improved Speed**: The warning ensures users understand the trade-off between visualization and speed, maintaining efficiency.  
- **Consistency**: Aligns plotting limits across different functions, reducing confusion for developers and users.  

This update makes validation more informative while balancing performance. 🌟

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting this 🚀 PR to the `ultralytics/ultralytics` repository! 

To ensure a smooth review process and integration of your contribution, please review the following checklist:

- ✅ **Define a Purpose**: Your PR description looks great! Be sure to link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues) (such as #19790, which you've correctly referenced). Ensure your commit messages are clear and aligned with our conventions.
- ✅ **Synchronize with Source**: Double-check that your branch is synchronized with the `main` branch of the repository. If your branch is outdated, you can update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify that all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address them to ensure the codebase remains stable.
- ✅ **Update Documentation**: If this change affects user workflows or introduces new functionality, kindly ensure the relevant [documentation](https://docs.ultralytics.com/) is updated.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes. Confirm that all tests pass successfully to maintain the integrity of the project.
- ✅ **Sign the CLA**: If this is your first contribution to Ultralytics, please sign our [Contributor License Agreement (CLA)](https://docs.ultralytics.com/help/CLA/) by commenting "I have read the CLA Document and I sign the CLA" in this thread.
- ✅ **Minimize Changes**: Focus on making only the necessary changes to address the issue or implement the feature. _"Perfection is achieved, not when there is nothing more to add, but when there is nothing left to take away."_ — Antoine de Saint-Exupéry

For further details, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). If you have any questions or need clarification, feel free to ask in this thread. 

This is an automated response, but rest assured that an Ultralytics engineer will review your PR and provide additional feedback soon. Thank you for contributing to Ultralytics! 🚀✨

**Comment by glenn-jocher** [comment]
@Y-T-G oh I didn't realize this was here. 15 seems very low. We should probably eliminate this cap or set it to max_det no?

Do you know what the speed implications are?

**Comment by glenn-jocher** [comment]
@Y-T-G also have this same limit hardcoded here:
https://github.com/ultralytics/ultralytics/blob/3896265a75253ff36c05d9b0a07589b42f5b05d8/ultralytics/models/yolo/segment/val.py#L297

**Comment by UltralyticsAssistant** [comment]
🎉 Huge thanks to @Y-T-G and @glenn-jocher for this fantastic contribution!  

As Henry Ford once said, *"Coming together is a beginning, staying together is progress, and working together is success."* This PR embodies that spirit, enhancing YOLO segmentation models with improved validation plotting and thoughtful consistency.  

Your work doesn’t just refine performance; it empowers users to visualize and understand their models better, a true leap toward innovation and usability. 🚀 We deeply appreciate your dedication and collaboration—together, we’re building something extraordinary! 🙌 Cheers to progress and teamwork! 👏

**Comment by Y-T-G** [comment]
@glenn-jocher I thought it was limited due to a memory issue. Yeah, 50 seems better.
